edgexfoundry/go-mod-core-contracts
# feat(data): Add new core-data reading API route and update ReadingClient

1. Add route for query readings by device name and time range while also allowing multiple resource names specified in request body as query criteria : `GET reading/device/name/{deviceName}/start/{start}/end/{end}`
2. Update ReadingClient to support above API

This is related to https://github.com/edgexfoundry/edgex-go/issues/3693

closes #678

Signed-off-by: Jude Hung <jude@iotechsys.com>







If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have opened a PR for the related docs change (if not, why?) The corresponding openapi doc will be updated through PR https://github.com/edgexfoundry/edgex-go/pull/3766
  <link to docs PR>

## Testing Instructions

This PR has to work with https://github.com/edgexfoundry/edgex-go/pull/3766
1. Modify the edgex-go's go.mod to replace the core-contracts lib
2. Run core-data
3. Add several events and readings with various resource name under the same device name.  Query readings through following API verify if expected readings are returned:
`GET reading/device/name/{deviceName}/start/{start}/end/{end}`

## New Dependency Instructions (If applicable)
